server/src/main/java/io/crate/execution/dml/ObjectIndexer.java

```diff
@@ -166,9 +171,10 @@ private void addNewColumns(Map<String, Object> value,
                 IndexType.PLAIN,
                 nullable,
                 storageSupport.docValuesDefault(),
-                -1,
+                position,
                 defaultExpression
             );
+            position--;
             onDynamicColumn.accept(newColumn);
```

Comment:
We recently had a fix https://github.com/crate/crate/pull/12754 for invalid column positions (including dynamic mapping updates).

That fix touched `TransportSchemaUpdateAction` and `FieldMappers` and since new indexer logic supposed to replace both, do I get it right, that column positions for dynamically added columns will be fully handled in the new indexers?

Reply:
It will still go through the `TransportSchemaUpdateAction.populateColumnPositions` to ensure the increment is atomic and there are no duplicates. But the (negative) numbering is importing to keep the column order of the statement. In `insert into tbl (a, b, c) ...` where `b` and `c` is new, `b` should appear before `c`.